### ISSUE TITLE
[x86] Changes to executable not updating in ghidra

### ISSUE THREAD

**Issue by greek-stasia** [NationalSecurityAgency/ghidra#1075]
**Describe the bug**
I have made some simple c programs to test this. But it seems when I import the same executable after I change a parameter and recompile it's not reflected in the analysis.

**To Reproduce**
1. Import executable and analyze in code browser
- Take note of a function you're about to change
2. Exit out of code browser
3. Go to program and add parameter recompile a before
4. Import executable again and analyze
- Take note of the same function as before

**Expected behavior**
Changes should be reflected in the new import

**Screenshots**

Before Change
![before_change](https://user-images.githubusercontent.com/24921719/65646213-5ab86d00-dfc8-11e9-9328-6b45e3f4cc9a.png)

Imported file after recompilation
![analyze ](https://user-images.githubusercontent.com/24921719/65646216-5d1ac700-dfc8-11e9-8527-80313fd4c7cb.png)

After change analysis
![after_change](https://user-images.githubusercontent.com/24921719/65646224-65730200-dfc8-11e9-8f30-f79d9f3231f7.png)

**Environment (please complete the following information):**
 - OS: Ubuntu 16.04.6 VM on macOS10.14.5
 - Java Version: 12.0.2
 - Ghidra Version:  9.0

**Additional context**
I'm wondering if there's some sort of cache or preference to toggle. Has this been mentioned before at all?  any ideas why this is happening?

**Comment by ryanmkurtz** [comment]
Could you do another test where you change the name of the function to see if the new version still has only one parameter but the new name?  That will help to narrow down the problem.

**Comment by greek-stasia** [comment]
Sure thing!

So here's I changed the name of the function above from toggle_lsb to just toggle.
![new_name](https://user-images.githubusercontent.com/24921719/65692514-b3c3e780-e040-11e9-9b55-cd9fe937286d.png)

And here I imported it after recompilation.
![name_change](https://user-images.githubusercontent.com/24921719/65692573-c9d1a800-e040-11e9-82fa-d153214ae287.png)

it doesn't seem to reflect the change

**Comment by ryanmkurtz** [comment]
Can you try using your 2nd parameter in the function (in a meaningful way)?  It might be optimizing it out.

**Comment by ghidra1** [comment]
Adding a function parameter which is never used will not be reflected in the decompiler auto-generated function signature.  The decompiler is limited to parameter detection based upon data flow analysis.  Alternatively, in the presence of debug information (e.g., DWARF, PDB) or mangled function names, analysis may be able to establish a function signature from such data.

**Comment by greek-stasia** [comment]
Not sure what you mean by "meaningful" since it is a relative and subjective term. 
So here I didn't change the function signature but incremented the second parameter:
![increment](https://user-images.githubusercontent.com/24921719/65721573-46cc4400-e078-11e9-931a-948ea921bb06.png)

Notice it doesn't seem to show the new change. But again, I recompiled it and imported it as a new file Though the function does get called by another function which then is called in main.

Then I decided to change the return type, hence, changing the function signature and then it seems to reflect.
![return_type](https://user-images.githubusercontent.com/24921719/65721741-9ca0ec00-e078-11e9-893d-4c7cf82a3591.png)


I appreciate the detailed description @ghidra1  .
